flink-formats/flink-csv/src/test/java/org/apache/flink/formats/csv/TableCsvFormatITCase.java

```diff
@@ -165,30 +269,87 @@ private static String formatSqlTimestamp(long timestamp) {
-        properties.put("format", "csv");
-
-        createTestSourceTable(tableName, fieldNameAndTypes, null, properties);
+        tableEnv.createTemporaryTable(
+                tableName,
+                TableDescriptor.forConnector(FileSystemTableFactory.IDENTIFIER)
+                        .option(FileSystemConnectorOptions.PATH, sourceFile.getAbsolutePath())
+                        .format(CsvCommons.IDENTIFIER)
+                        .schema(schema)
+                        .build());
     }
 
-    private File createSinkTable(String tableName, String... fieldNameAndTypes) throws IOException {
+    private File createSinkTable(String tableName, Schema schema) throws IOException {
         File sinkPath = TEMPORARY_FOLDER.newFolder();
 
-        Map<String, String> properties = new HashMap<>();
-        properties.put("connector", "filesystem");
-        properties.put("path", sinkPath.getAbsolutePath());
-        properties.put("format", "csv");
-        properties.put("csv.disable-quote-character", "true");
+        tableEnv.createTemporaryTable(
+                tableName,
+                TableDescriptor.forConnector(FileSystemTableFactory.IDENTIFIER)
+                        .option(FileSystemConnectorOptions.PATH, sinkPath.getAbsolutePath())
+                        .option("csv.disable-quote-character", "true")
+                        .format(CsvCommons.IDENTIFIER)
+                        .schema(schema)
+                        .build());
 
-        createTestSinkTable(tableName, fieldNameAndTypes, null, properties);
         return sinkPath;
     }
+
+    private void createTestValuesSourceTable(
+            String tableName,
+            List<Row> data,
+            Schema schema,
+            Map<String, String> extraProperties,
+            @Nullable String... partitionFields) {
+
+        String dataId = TestValuesTableFactory.registerData(data);
+        Map<String, String> properties = new HashMap<>();
+        properties.put("data-id", dataId);
+        properties.put("bounded", "true");
+        properties.put("disable-lookup", "true");
+        properties.putAll(extraProperties);
+
+        TableDescriptor.Builder descriptor =
+                TableDescriptor.forConnector("values")
+                        .schema(schema)
+                        .partitionedBy(partitionFields);
+
+        for (Map.Entry<String, String> entry : properties.entrySet()) {
+            descriptor.option(entry.getKey(), entry.getValue());
+        }
+
+        tableEnv.createTemporaryTable(tableName, descriptor.build());
+    }
+
+    private void assertResult(List<String> expected, File resultFile) throws IOException {
+        List<String> actual = readLines(resultFile);
+        assertResult(expected, actual);
+    }
+
+    private void assertResult(List<String> expected, List<String> actual) {
+        Collections.sort(expected);
+        Collections.sort(actual);
+        assertThat(actual).isEqualTo(expected);
+    }
```

Comment:
Using assertJ could be reduced to `assertThat(actual).hasSameElementsAs(expected);`

Reply:
Good catch, that was a residue from the JsonPlanTestBase.